### ISSUE TITLE
Translations

### ISSUE THREAD

**Issue by harryhaaren** [openAVproductions/openAV-manual#1]
A place-holder for discussing how to host multiple-language manuals.

**Comment by rcaneill** [comment]
What is the easiest thing to do: create new files named fr-beginner.rst or looking for a way to implement the translated versions directly in the same file (as the english version)?

**Comment by harryhaaren** [comment]
Yes a separate file is better, your suggestion is perfect, fr-name is good. Thanks!

**Comment by rcaneill** [comment]
I forked the manual here to work on the french translation: [https://github.com/rcaneill/manual](https://github.com/rcaneill/manual). If they are any frenchies, feel free to help me!

**Comment by georgkrause** [comment]
I started a german Version here: https://github.com/georgkrause/manual/tree/german 
I didnt spend that much time right now, but reviews/additions welcome ;)

**Comment by coderkun** [comment]
@georgkrause, I would like to add some German translation for luppp as well. @harryhaaren, please let me know when the “multi-language” files are in place (as mentioned in #9).

**Comment by georgkrause** [comment]
@coderkun you are welcome! simply give me a PR, i will merge it and if we finished we can give a PR to harry. In that way you don't need to worry about file names, i will do this ;)

**Comment by coderkun** [comment]
@georgkrause, sounds good. Will file a PR for German additions.

**Comment by georgkrause** [comment]
@coderkun i updated the file names. if you start working on another file, please put the prefix before (like de-artyfx.rst)

**Comment by coderkun** [comment]
@georgkrause, do you mind pulling the latest changes from upstream which contain some changes for Luppp into your fork? It would be good to have these changes before starting the translation.

**Comment by georgkrause** [comment]
@coderkun should be done.

**Comment by harryhaaren** [comment]
Hey All,

We need to find a good workflow for these translations - I have investigated the various options, and I think the best fit is to have each language on a seperate branch in git. The work flow is approx as follows:

1) New content goes on the english version - this should always be kept up to date, so we have an "official" version, which can be duplicated to the various langauges. `master` branch will contain the english docs
2) Each other language (`fr` and `de` in progress now) are based on the `master` branch, with commits on top to translate the content. Each `fr` or `de` branch will have its own files commited to index.rst, so running `make html` will generate the translated docs (and not the english version). 
3) To update the translation, the language branch can merge in the latest from `master` and update content as required.

That's the best I can think of for now - anybody see obvious issues with this?

I've pushed a `fr` branch for testing purposes, `fr` folks, please have a play with it and see. @georgkrause if you file a PR I can push a `de` branch too :)

Cheers guys! -H

**Comment by harryhaaren** [comment]
Update: the languages will become available at `openavproductions.com/doc/<language>/`
For example, `fr` is available here:
http://openavproductions.com/doc/fr/index.html

Yes the index.html is currently mandatory, I need to fix the url-rewriting and then `/doc/fr/`will be all that is required!

**Comment by georgkrause** [comment]
Sounds good, i will prepare a pr today

**Comment by georgkrause** [comment]
In fact since there is no finished german file, it will take some more time to file a PR. But @coderkun has nearly translated the Luppp file and i will work on the artyfx in my spare time.
